### DISCUSSION TITLE
macos 11.3 upgrade causes mytop to fail with ListUtil.c: loadable library and perl binaries are mismatched

### DISCUSSION THREAD

**Discussion by kylezeeuwen** [Homebrew/discussions#1381] [Casks] [answered]
Hello,

I'm also dpocumenting and looking for help over here : https://stackoverflow.com/questions/67308565/mytop-on-big-sur-stopped-working-after-os-update

Timeline:
* Apr 27 : all is well running macos 11.2.X, mytop runs like a pro, showing my SQL things
* Apr 28: install macos 11.3
* mytop now fails with:

```
> mytop
ListUtil.c: loadable library and perl binaries are mismatched (got handshake key 0xc500080, needed 0xc400080)

> which mytop
/usr/local/bin/mytop

> ls -la /usr/local/bin/mytop
lrwxr-xr-x  1 username  admin  33  9 Dec 10:24 /usr/local/bin/mytop -> ../Cellar/mytop/1.9.1_8/bin/mytop
```

Things I've done so far to try to fix:

```
* brew update
* xcode-select --install (wait 5 hours)
* brew upgrade (upgrades 19 things)
* brew remove mytop; brew install mytop
```

**Comment by carlocab** [comment, marked as answer]
This was caused by Big Sur 11.3 switching the default `perl` to 5.30. It used to be 5.28, and that's the version that `mytop` expects to find at `/usr/bin/perl`. See https://github.com/Homebrew/brew/issues/10127.

In the meantime, try `brew reinstall -s mytop` to rebuild `mytop` against the new version of system `perl`.

**Reply by kylezeeuwen** [reply]
this worked 🙏
